### PR TITLE
Allow SC reconnect after signing create_tx

### PR DESCRIPTION
**Pull request by uwiger**
See issue #3169 

Some refactoring of the sc suite, to allow for partial setup of a channel, pausing before the create_tx has been mined onto the chain.

In local test runs, `aehttp_sc_SUITE` fails with errors seemingly related to contract cache issues. This seems separate from the present fix.

### CODE REVIEW

File: apps/aechannel/src/aesc_fsm.erl [hunk 1]
```diff
@@ -3184,8 +3215,19 @@ watch_req() ->
 on_chain_id(#data{on_chain_id = ID} = D, _) when ID =/= undefined ->
     {ID, D};
 on_chain_id(D, SignedTx) ->
-    {ok, PubKey} = aesc_utils:channel_pubkey(SignedTx),
-    {PubKey, D#data{on_chain_id = PubKey}}.
+    try aesc_utils:channel_pubkey(SignedTx) of
+        {ok, PubKey} ->
+            {PubKey, D#data{on_chain_id = PubKey}}
+    ?CATCH_LOG(_E)
+        if D#data.strict_checks == false ->
```
Comment (velzevur) [inline apps/aechannel/src/aesc_fsm.erl]: `strict_checks` is for AliceFSM, in the case Alice tries cheating Bob. That indeed would be a test scenario. I am wondering if we get here at all.

Comment (velzevur) [reply apps/aechannel/src/aesc_fsm.erl]: Yep, the `check_incorrect_create` goes right there, this is indeed needed

**Comment by velzevur** [comment]
Since a new state had been added, please update the UML

**Comment by uwiger** [comment]
Superseded by #3224

**Pull request closed without merging**